icaruscode/Decode/DaqDecoderICARUSTPCwROI_module.cc

```diff
@@ -585,20 +585,15 @@ void DaqDecoderICARUSTPCwROI::processSingleFragment(size_t
         {
             mf::LogInfo(fLogCategory) << "==> Found board/boardSlot mismatch, crate: " << crateName << ", board: " << board << ", boardSlot: " << boardSlot << " channelPlanePair: " << fChannelMap->getChannelPlanePair(boardIDVec[board]).front().first << "/"  << fChannelMap->getChannelPlanePair(boardIDVec[board]).front().second << ", slot: " << channelPlanePairVec[0].first << "/" << channelPlanePairVec[0].second;
         }
-
-        // Get the pointer to the start of this board's block of data
-        const icarus::A2795DataBlock::data_t* dataBlock = physCrateFragment.BoardData(board);
-
         // Copy to input data array
         for(size_t chanIdx = 0; chanIdx < nChannelsPerBoard; chanIdx++)
         {
            icarus_signal_processing::VectorFloat& rawDataVec = channelArrayPair.second[chanIdx];
+           for (size_t tick = 0; tick < nSamplesPerChannel; ++tick)
+             rawDataVec[tick] = physCrateFragment.adc_val(board, chanIdx, tick);
```

Comment:
Does this new method preserve the sign of the ADC value (note the "-" sign in the removed code)

Reply:
Ah, no it doesn't. The function `adc_val(board, chanIdx, tick)` returns the ADC as a positive value. It's not entirely clear to me why we want to store the ADCs as negative values, but since I don't want to introduce unintended behaviors I can add in the `-`.

Reply:
I notice `icaruscode/Decode/DecoderTools/TPCDecoder_tool.cc`, unlike the other decoder tools, doesn't have the `-`. Should I leave that one as is?

Reply:
When we saw the first waveforms from the TPC they had negative polarity which is not the way that “standard” LArSoft tools were expecting to see things - hence the minus sign you saw on the decoder tools we actually use. If the new function is return a positive polarity waveform then all is good. Otherwise we will need the minus sign.

Reply:
So I don't know about the polarity, but the values being read out of the fragments are negative when I use `rawDataVec[tick] = -dataBlock[chanIdx + tick * nChannelsPerBoard]`, but are positive when I use `rawDataVec[tick] = physCrateFragment.adc_val(board, chanIdx, tick)`. I tested this by running `stage0_run2_icarus.fcl` on `data_dl1_fstrmOffBeamBNBMINBIAS_run11418_6_20240118T024159.root` with the following debug check:
```
         // Copy to input data array
         for(size_t chanIdx = 0; chanIdx < nChannelsPerBoard; chanIdx++)
         {
            icarus_signal_processing::VectorFloat& rawDataVec = channelArrayPair.second[chanIdx];
            for (size_t tick = 0; tick < nSamplesPerChannel; ++tick)
            {
              rawDataVec[tick] = -physCrateFragment.adc_val(board, chanIdx, tick);
              float advValue = physCrateFragment.BoardData(board)[chanIdx + tick * nChannelsPerBoard];
              if (tick == 7)
                mf::LogDebug(fLogCategory)
                  << "FragmentID " << std::hex << fragmentID << std::dec << ", Crate " << crateName<< ",  board " << board 
                  << ", channel " << chanIdx << ", sample " << tick << " has ADC value " << advValue << " | " << rawDataVec[tick];
            }
 
           // Keep track of the channel
           channelArrayPair.first[chanIdx] = channelPlanePairVec[chanIdx];
         }
```
Which shows the methods give opposite signs:
```FragmentID 1700, Crate WW11,  board 6, channel 2, sample 7 has ADC value 2020 | -2020```
So if we want to preserve the current function of the Decoder Tools we should sill have the `-`.

Reply:
If I understand the last comment then we need to introduce a "-" sign to preserve current behavior. This is important for the deconvolution since it expects to see waveforms with polarity as it currently returned. Is it easy (and possible) to overlay a waveform near a signal for the two methods to make sure we have things right?